core/jazz_ui/e2e/04-apiservices/apiservices.e2e-spec.ts

```diff
@@ -0,0 +1,316 @@
+/*
+ *Copyright 2016-2017 T Mobile, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); You may not use
+ * this file except in compliance with the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { browser, element, by, protractor, $ } from 'protractor';
+import { Jazz } from '../page-objects/jazzservices.po';
+import { CONFIGURATIONS } from '../../src/config/configuration';
+import { Timeouts, Browser } from 'selenium-webdriver';
+import { SSL_OP_SSLEAY_080_CLIENT_DH_BUG } from 'constants';
+
+const twok = 2000;
+const fivek = 5000;
+const tenk = 10000;
+const fifteenk = 15000;
+const twentyk = 20000;
+const thirtyk = 30000;
+const sixtyk = 60000;
+
+const timeOutHigh = 180000;
+const emailId = CONFIGURATIONS.optional.general.e2e.EMAIL_ID;
+const config = CONFIGURATIONS.optional.general.e2e;
+
+describe('Overview', () => {
+  let jazzServices_po: Jazz;
+
+  const EC = protractor.ExpectedConditions;
+  let winhandle;
+  let servicename;
+  let test;
+
+  beforeAll(() => {
+    jazzServices_po = new Jazz();
+
+  });
+
+  function createservice(servicename) {
+    jazzServices_po.getServiceName().sendKeys(servicename);
+    jazzServices_po.getNameSpace().sendKeys('jazztest');
+    jazzServices_po.getServiceDescription().sendKeys('Testing');
+  }
+
+  function serviceapprover() {
+    browser.driver.sleep(fivek);
+    jazzServices_po.getSubmit().click();
+    fluentwaittry(jazzServices_po.getDone(), tenk);
+    jazzServices_po.getDone().click();
+  }
+  function waitForSpinnerDisappear() {
+    browser.wait(EC.not(EC.visibilityOf(jazzServices_po.getSpinner())), 180000);
+  }
+  function waitForMetricsSpinner() {
+    browser.wait(EC.not(EC.visibilityOf(jazzServices_po.getMetricsSpinner())), 180000);
+  }
+  function fluentwaittry(ele, t) {
+    browser.manage().timeouts().implicitlyWait(0);
+    browser.wait(function () {
+      browser.sleep(t);
+      return ele.isDisplayed()
+        .then(
+          function (text) {
+            return text;
+
+          },
+          function (error) {
+            browser.refresh();
+            return error;
+          });
+    }, 180 * 1000);
+  }
+
+  function waitforservice(ele, t) {
+    browser.manage().timeouts().implicitlyWait(0);
+    browser.wait(function () {
+      browser.sleep(t);
+      return ele.isDisplayed()
+        .then(
+          function (text) {
+            return text;
+
+          },
+          function (error) {
+            browser.refresh();
+            return false;
+          });
+    }, 240 * 1000);
+  }
+
+  function refreshbutton(ele, t) {
```

Comment:
frequently used functions should be moved to a util file

Reply:
removed